### PR TITLE
[sdks] Transition for storing the llvm version info in llvm/SUBMODULE…

### PR DESCRIPTION
**Pull request by vargaz**
…S.json and using the makefiles/scripts there to clone the repo etc.

### CODE REVIEW

**Comment by vargaz** [comment]
There is a dependency problem since llvm/Makefile is created by configure, and configure needs an already built llvm. So we will probably need to make llvm/Makefile a normal makefile.

**Comment by luhenry** [comment]
I'm fine with making `llvm/Makefile` a normal Makefile since there are no sources or anything in there other than the logic to fetch llvm and build it.

**Comment by vargaz** [comment]
So making llvm/Makefile a normal makefile is a little bit messy, it might be a better approach to get rid of it completely, and move the logic there to the sdk makefiles which are already normal makefiles.

**Comment by luhenry** [comment]
@vargaz you are saying "configure needs an already built llvm", why is that? And instead of having these dependencies at configure-time, could we have them at compile-time instead?

**Comment by vargaz** [comment]
configure needs to run the llvm-config script to obtain the list of include dirs, libraries, gcc flags, etc. Running it at compile time would be kinda complicated with automake.

**Comment by luhenry** [comment]
Ok, let's get it to depend on `sdks/` then. I don't have a strong opinion on which way we should go, so it's up to you.

File: sdks/builds/llvm.mk [hunk 1]
```diff
@@ -1,9 +1,16 @@
 
 LLVM_SRC?=$(TOP)/sdks/builds/toolchains/llvm
 
-$(TOP)/sdks/builds/toolchains/llvm:
```
Comment (luhenry) [inline sdks/builds/llvm.mk]: Why don't we just keep the existing code since it does the same thing?

Comment (vargaz) [reply sdks/builds/llvm.mk]: The code in versions.mk reads the hashes from SUBMODULES.json and makes sure that the proper version is checked out if the hash changes. Its also used by acceptance-tests/ so there is code sharing.

**Comment by vargaz** [comment]
Superseded by https://github.com/mono/mono/pull/8792

**Pull request closed without merging**